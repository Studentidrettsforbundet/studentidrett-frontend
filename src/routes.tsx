import React from 'react';
import { Provider } from 'react-redux';
import { Route, BrowserRouter as Router, Switch } from 'react-router-dom';
import App from './App';
import RegionPage from './pages/regionPage';
import CityPage from './pages/cityPage';
import SportPage from './pages/sportPage';
import ClubPage from './pages/clubPage';
import GroupPage from './pages/groupPage';
import QuestionnairePage from './pages/QuestionnairePage';
import ResultPage from './pages/ResultPage';
import TeamPage from './pages/teamPage';
import store from './store/store';

const createRoutes = () => (
    <Provider store={store}>
        <Router>
            <div className="page">
                <Switch>
                    <Route exact path="/" component={App} />
                    <Route exact path="/questionnaire" component={QuestionnairePage} />
                    <Route exact path="/questionnaire/result" component={ResultPage} />
                    <Route exact path="/Regions" component={RegionPage} />
                    <Route exact path="/:City" component={CityPage} />
                    <Route exact path="/:City/:Club" component={ClubPage} />
                    <Route exact path="/:City/Sport/:Sport" component={SportPage} />
                    <Route exact path="/:City/:Club/:Group" component={GroupPage} />
                </Switch>
            </div>
        </Router>
    </Provider>
);

export default createRoutes;
