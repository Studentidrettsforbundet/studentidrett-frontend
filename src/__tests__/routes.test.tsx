import React from 'react';
import renderer from 'react-test-renderer';
import { Link, MemoryRouter } from 'react-router-dom';
import routes from '../index';


describe('routes.tsx', () => {
    test("renders home Link correctly", () => {
        const tree = renderer
            .create(<MemoryRouter>
                <Link to={"/Regions"} key={0} className={'unstyled_link'}/>
            </MemoryRouter>);
        expect(tree).toMatchSnapshot();
    });
    test("renders region Link correctly", () => {
        const tree = renderer
            .create(<MemoryRouter>
                <Link to={"/NordNorge"} key={0} className={'unstyled_link'} />
            </MemoryRouter>);
        expect(tree).toMatchSnapshot();
    });
    test("renders sport Link correctly", () => {
        const tree = renderer
            .create(<MemoryRouter>
                <Link to={"NordNorge/Fotball"} key={0} className={'unstyled_link'} />
            </MemoryRouter>);
        expect(tree).toMatchSnapshot();
    });
    test("renders club Link correctly", () => {
        const tree = renderer
            .create(<MemoryRouter>
                <Link to={"MidtNorge/Fotball/NTNUIFotball"} key={0} className={'unstyled_link'} />
            </MemoryRouter>);
        expect(tree).toMatchSnapshot();
    });
});

describe('index.tsx', () => {
    test("renders correctly", () => {
        const tree = renderer
            .create(<MemoryRouter>
                routes
            </MemoryRouter>);
        expect(tree).toMatchSnapshot();
    });
});