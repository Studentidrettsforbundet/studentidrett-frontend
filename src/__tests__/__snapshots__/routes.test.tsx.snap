// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index.tsx renders correctly 1`] = `"routes"`;

exports[`routes.tsx renders club Link correctly 1`] = `
<a
  className="unstyled_link"
  href="/MidtNorge/Fotball/NTNUIFotball"
  onClick={[Function]}
/>
`;

exports[`routes.tsx renders home Link correctly 1`] = `
<a
  className="unstyled_link"
  href="/Regions"
  onClick={[Function]}
/>
`;

exports[`routes.tsx renders region Link correctly 1`] = `
<a
  className="unstyled_link"
  href="/NordNorge"
  onClick={[Function]}
/>
`;

exports[`routes.tsx renders sport Link correctly 1`] = `
<a
  className="unstyled_link"
  href="/NordNorge/Fotball"
  onClick={[Function]}
/>
`;
